File: ToyInterpreter/ToyInterpreter/Repository/MemRepo.cs
using System.IO;
using System.Runtime.Serialization;
using System.Runtime.Serialization.Formatters.Binary;
using System;
using ToyInterpreter.Model;

namespace ToyInterpreter.Repository
{
	public class MemRepo
	{
		private ProgState ps;
		public MemRepo (ProgState ps) {
			this.ps = ps;
		}

		public bool SerializeToFile() {
			try {
				FileStream fs = File.Create("toyInterpreter.txt");

				BinaryFormatter bformatter = new BinaryFormatter();

				bformatter.Serialize(fs, ps);

				fs.Close();
				return true;
			}
			catch (SerializationException e) {
				Console.WriteLine("ERROR: could not write: " + e.Message);
			}
			catch (IOException e) {
				Console.WriteLine("ERROR: could not write: " + e.Message);
			}
			return false;
		}

		public ProgState SerializeFromFile() {
			try {
				FileStream fs = File.OpenRead("toyInterpreter.txt");
				BinaryFormatter bformatter = new BinaryFormatter();

				ps = (ProgState) bformatter.Deserialize(fs);

				fs.Close();
				return ps;
			}
			catch (SerializationException e) {
				Console.WriteLine("ERROR: could not read: " + e.Message);
			}
			catch (IOException e) {
				Console.WriteLine("ERROR: could not read: " + e.Message);
			}

			return null;
		}
	}
}

File: ToyInterpreter 2/src/Repository/MemRepo.java
package Repository;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

import Model.ProgState;

public class MemRepo {
	ProgState ps;
	public MemRepo(ProgState ps) { this.ps=ps;}
	public void serializeToFile() {
        try {
            FileOutputStream fileOut = new FileOutputStream("toyInterpreter.txt");
            ObjectOutputStream out = new ObjectOutputStream(fileOut);

            out.writeObject(ps);

            out.close();
            fileOut.close();
        }
        catch (IOException e) {
            System.out.println("ERROR: could not write: " + e.getMessage());
            e.printStackTrace();
        }
    }
	
    public ProgState serializeFromFile() {
        try {
            FileInputStream fileIn = new FileInputStream("toyInterpreter.txt");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            ps = (ProgState) in.readObject();
            in.close();
            fileIn.close();
            return ps;
        }
        catch (IOException e)
        {
            System.out.println("ERROR: could not read: " + e.getMessage());
            e.printStackTrace();
        }
        catch (ClassNotFoundException e)
        {
            System.out.println("ERROR: could not find class: " + e.getMessage());
            e.printStackTrace();
        }
        return null;
    }
}
